src/components/BaseLayerSwitcher/BaseLayerSwitcher.js

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable jsx-a11y/interactive-supports-focus */
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FaChevronCircleLeft } from 'react-icons/fa';
+import Layer from '../../layers/Layer';
+import img from '../../images/baselayer/osm.baselayer.png';
+
+import './BaseLayerSwitcher.scss';
+
+const propTypes = {
+  /**
+   * An array of react-spatial layers.
+   */
+  layers: PropTypes.arrayOf(PropTypes.instanceOf(Layer)).isRequired,
+
+  /**
+   * Object containing relative paths to the base layer images.
+   */
+  layerImages: PropTypes.objectOf(PropTypes.string),
+
+  /**
+   * CSS class to apply on the container.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Alternative text rendered if layer images can't be loaded
+   */
+  altText: PropTypes.string,
+
+  /**
+   * Button titles.
+   */
+  titles: PropTypes.shape({
+    button: PropTypes.string,
+    openSwitcher: PropTypes.string,
+    closeSwitcher: PropTypes.string,
+  }),
+};
+
+const defaultProps = {
+  className: 'rs-base-layer-switcher',
+  altText: 'Source not found',
+  titles: {
+    button: 'Base layers',
+    openSwitcher: 'Open Baselayer-Switcher',
+    closeSwitcher: 'Close Baselayer-Switcher',
+  },
+  layerImages: undefined,
+};
+
+const getVisibleLayer = layers => {
+  return layers.find(layer => layer.getVisible());
+};
+
+const getNextImage = (currentLayer, layers, layerImages) => {
+  const currentIndex = layers.indexOf(
+    layers.find(layer => layer === currentLayer),
+  );
+  const nextIndex = currentIndex + 1 === layers.length ? 0 : currentIndex + 1;
+  return layerImages[nextIndex];
+};
+
+function BaseLayerSwitcher({
+  layers,
+  layerImages,
+  altText,
+  className,
+  titles,
+}) {
+  const baseLayers = layers.filter(layer => layer.getIsBaseLayer());
+  const [switcherOpen, setSwitcherOpen] = useState(false);
+  const [isClosed, setIsClosed] = useState(true);
+  const [currentLayer, setCurrentLayer] = useState(getVisibleLayer(baseLayers));
+
+  /* Images are loaded from props if provided, fallback from layer */
+  const images = layerImages
+    ? Object.keys(layerImages).map(layerImage => layerImages[layerImage])
+    : baseLayers.map(layer => layer.previewImage);
+
+  const openClass = switcherOpen ? ' rs-open' : '';
+  const closedClass = isClosed ? ' rs-closed' : '';
+
+  const onButtonSelect = (open, layer) => {
+    if (!open) {
+      setSwitcherOpen(true);
+      return;
+    }
+    setCurrentLayer(layer);
+    layer.setVisible(true);
+    setSwitcherOpen(false);
+  };
+
+  const setTabIndex = (closed, buttonIndex) => {
+    if (closed) {
+      return buttonIndex === 0 ? '0' : '-1';
+    }
+    return '0';
+  };
+
+  if (!baseLayers || baseLayers.length < 2) {
+    return null;
+  }
```

Comment:
I think this 'if' must be moved after all the useEffect function , because hook must  always be executed in the same order. It's surprising eslint is not complaining.

Reply:
Thanks for pointing it out